src/dmd/dstruct.d

```diff
@@ -607,6 +607,53 @@ extern (C++) class StructDeclaration : AggregateDeclaration
     {
         return index < numArgTypes() ? (*argTypes.arguments)[index].type : null;
     }
+
+    final bool hasNonDisabledCtor()
+    {
+        static extern (C++) class HasNonDisabledCtorVisitor : Visitor
+        {
+            bool result;
+
+            this()
+            {
+                result = false;
+            }
+
+            alias visit = Visitor.visit;
+
+            override void visit(CtorDeclaration cd)
+            {
+                if (!(cd.storage_class & STC.disable))
+                {
+                    result = true;
+                }
+            }
+
+            override void visit(TemplateDeclaration td)
+            {
+                result = true;
+            }
+
+            override void visit(OverloadSet os)
+            {
+                for (size_t i = 0; i < os.a.dim; i++)
+                {
+                    os.a[i].accept(this);
+                }
+            }
+        }
+
+        if (ctor)
+        {
+            auto v = new HasNonDisabledCtorVisitor();
+            ctor.accept(v);
+            return v.result;
+        }
+        else
+        {
+            return false;
+        }
```

Comment:
```suggestion
        if (!ctor)
            return false;
        scope v = new HasNonDisabledCtorVisitor();
        ctor.accept(v);
        return v.result;
```

In general, we prefer to use early returns like this, to get the obvious case "out of the way" and simplify the flow, as well as reduce the amount of unneeded indentation.